### DISCUSSION TITLE
Dual arm control with a single robot

### DISCUSSION THREAD

**Discussion by supersglzc** [NVlabs/curobo#337] [Software Q&A]
If it’s not a bug, please use discussions: https://github.com/NVlabs/curobo/discussions

Please provide the below information in addition to your issue:

1. cuRobo installation mode (choose from [python, isaac sim, docker python, docker isaac sim]): isaac sim
2. python version: 3.10.14
3. Isaac Sim version (if using): 4.0.0

------
Hi, I have tried the Multi-Arm Reacher example and it works fine. Basically it defines two goals by specifying the link names in yml. However, when I tried the same thing on my dual-arm robot, in which two arms are connected by a shared torso, it often gives me: Plan did not converge to a solution. I am sure that those poses are easy to reach. If I disable one side of the arm joints to fixed, everything is smooth. 

I suppose this is the issue of the connected body, is there any idea how I can resolve this? I don't want to create two separate robots instances and plan separately...

Thanks!

**Comment by bornaparo** [comment]
hi @supersglzc , could you please share your robot's `.urdf` and `.xrdf` files? I am having similar problem with baxter robot when using isaac_ros_cumotion moveit plugin, but in my case it doesn't successfully plan even for just one arm ([more details in discussion](https://github.com/NVlabs/curobo/discussions/448)).

**Comment by balakumar-s** [comment]
Multi-arm reaching does not work as well as single arm reaching. You could start with checking if multi-target IK works for your robot. If it doesn't work well, try increasing the number of ik iterations here (300-400 might be a good number to start with): https://github.com/NVlabs/curobo/blob/2fbffc35225398cf9d5f382804faa9de2608753b/src/curobo/content/configs/task/gradient_ik_autotune.yml#L88

**Reply by Alice-ly** [reply]
![1740130383953](https://github.com/user-attachments/assets/0939ed2b-6952-4ecc-81f6-ba5294ef6e67)
Hello, I want to know what the link_names mean,  I have also using curobo in a dual-arm robot, for which I want to respectively control the left arm or the right arm due to the planning group set in plan request. And I use xrdf file now instead, I want to ask whether there is a convenient way to satisfy this feature.
